### PR TITLE
Log error when Deployment can't be created due to missing Secret

### PR DESCRIPTION
**Pull request by asymmetric**
It's an error so we should use `Error`.

Also, I think we don't need to check the resource for `== nil`, as I don't think Kubernetes would deliver us a `nil` struct if there wasn't an `error`.

### CODE REVIEW

File: pkg/habitat/controller/controller.go [hunk 1]
```diff
@@ -402,20 +402,15 @@ func (hc *HabitatController) newDeployment(sg *crv1.ServiceGroup) (*appsv1beta1.
 		// Let's make sure our secret is there before mounting it.
 		secret, err := hc.config.KubernetesClientset.CoreV1().Secrets(apiv1.NamespaceDefault).Get(sg.Spec.Habitat.Config, metav1.GetOptions{})
 		if err != nil {
-			return nil, err
-		}
-
-		if secret == nil {
-			// If we can't find the secret we should just return the base deployment.
-			level.Warn(hc.logger).Log("msg", "Secret could not be mounted as it did not exist.")
+			level.Error(hc.logger).Log("msg", "Could not find Secret containing service configuration")
```
Comment (LiliC) [inline pkg/habitat/controller/controller.go]: Can we maybe mention the Secret name in the log?

**Comment by asymmetric** [comment]
PTALA.

File: pkg/habitat/controller/controller.go [hunk 2]
```diff
@@ -402,20 +396,15 @@ func (hc *HabitatController) newDeployment(sg *crv1.ServiceGroup) (*appsv1beta1.
 		// Let's make sure our secret is there before mounting it.
 		secret, err := hc.config.KubernetesClientset.CoreV1().Secrets(apiv1.NamespaceDefault).Get(sg.Spec.Habitat.Config, metav1.GetOptions{})
 		if err != nil {
-			return nil, err
-		}
-
-		if secret == nil {
-			// If we can't find the secret we should just return the base deployment.
-			level.Warn(hc.logger).Log("msg", "Secret could not be mounted as it did not exist.")
+			level.Error(hc.logger).Log("msg", "Could not find Secret containing service configuration", "name", sg.Spec.Habitat.Config)
```
Comment (LiliC) [inline pkg/habitat/controller/controller.go]: Looking at it again, I don't think we need to log this. As we already return an error and later on log it:
https://github.com/kinvolk/habitat-operator/blob/1a3261ebbeffcc1dbcc030371571213d544ff5ce/pkg/habitat/controller/controller.go#L146

wdyt?

**Comment by asymmetric** [comment]
PTALA.

**Pull request merged**